src/program/mod.rs

```diff
@@ -155,11 +164,48 @@ impl Program {
                 },
                 Instruction::Delay(delay) => delay.qubits.clone(),
                 Instruction::Fence(fence) => fence.qubits.clone(),
+                Instruction::Capture(capture) => capture.frame.qubits.clone(),
+                Instruction::Pulse(pulse) => pulse.frame.qubits.clone(),
+                Instruction::RawCapture(raw_capture) => raw_capture.frame.qubits.clone(),
                 _ => vec![],
             })
             .collect::<HashSet<_>>()
     }
 
+    /// Simplify this program into a new [Program] which contains only instructions
+    /// and definitions which are executed; effectively, perform dead code removal.
+    ///
+    /// Removes:
+    /// - All calibrations, following calibration expansion
+    /// - Frame definitions which are not used by any instruction such as `PULSE` or `CAPTURE`
+    /// - Waveform definitions which are not used by any instruction
+    ///
+    /// When a valid program is simplified, it remains valid.
+    pub fn into_simplified(&self) -> Result<Self> {
+        let mut expanded_program = self.expand_calibrations()?;
+        expanded_program.calibrations = CalibrationSet::default();
```

Comment:
A comment on why you replace the calibrations would be nice. As someone new to this part of the code, it looks like you are setting the calibrations, only to reset them.

Reply:
@Shadow53 It doesn't _set_ calibrations -- it _expands_ them, and then removes the calibration definition. Think of it as inline a function body: the "calibration" is the function, and instructions that reference the calibration are replaced with the calibration body.